### DISCUSSION TITLE
从Stencil拖入Node，放置在Edge上，实现嵌入Edge的效果

### DISCUSSION THREAD

**Discussion by zzyyu** [antvis/X6#3252] [Q&A]
想要实现Node自动嵌入Edge的效果，请问有什么好的实现方式？
拖入：
 ![image](https://user-images.githubusercontent.com/33147972/219562904-fd6ff28d-0dcb-4bc8-a180-093c59acdbf0.png)

变成：
![image](https://user-images.githubusercontent.com/33147972/219563148-2240ca51-d9d3-4380-a5c5-0f69df200d64.png)

**Comment by NewByVector** [comment]
可以参考 [demo](https://codesandbox.io/s/yu-lian-xian-xiang-jiao-zi-dong-lian-jie-boh3jk)。

**Reply by zzyyu** [reply]
感谢您的demo，demo中使用了“node:moving”事件，但从Stencil拖动时，还处于拖拽阶段，无法触发该事件
我最终想要实现拖动中预览的效果，虽然可以通过“dnd”的“getDragNode”选项，获取到拖动的node对象，但是无法实时监控到拖动的位置，请问有好的方案吗

我想要实现的效果（未放置，但是能够预览放置效果）：
![image](https://user-images.githubusercontent.com/33147972/219595027-19cd0586-9596-4093-a304-2235068e13b8.png)

**Reply by zzyyu** [reply]
好像已经找到方案了，Stencil中是根据群组生成了对应的graph
```
this.stencil.graphs.群组名称.on("cell:mousemove", ({ e, x, y, cell, view }) => {
        console.log("cell:mousemove", cell);
      });
```
就可以获取到拖拽中node对应移动的位置了
